providers/hns/hns_roce_u_verbs.c

```diff
@@ -853,15 +877,30 @@ static int qp_exec_create_cmd(struct ibv_pd *pd,
 	cmd.log_sq_stride = qp->sq.wqe_shift;
 	cmd.log_sq_bb_count = hr_ilog32(qp->sq.wqe_cnt);
 
-	ret = ibv_cmd_create_qp(pd, &qp->ibv_qp, attr, &cmd.ibv_cmd,
-				sizeof(cmd), &resp.ibv_resp, sizeof(resp));
+	if (attr->comp_mask & CREATE_QP_EX2_COMP_MASK) {
```

Comment:
Thre is no reason to have this if, just always use ibv_cmd_create_qp_ex2() and always decleare the ex cmd structure

Reply:
Hi Jason, I have updated the patch. Is there anything that needs to be modified?